### PR TITLE
Fix Issue 19738 - std.range.choose assignment breaks @safe-ty

### PR DESCRIPTION
**Pull request by thewilsonator**
cc @WalterBright @dnadlinger

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -1434,6 +1434,36 @@ if (isInputRange!(Unqual!R1) && isInputRange!(Unqual!R2) &&
...
+            return this;
+        }
+    }
+    enum isDIP1000 = __traits(compiles, () @safe
+    {
+      int x;
+      int* p;
+      p = &x;
+    });
+    static if (isDIP1000)
```
Comment (dnadlinger) [inline std/range/package.d]: Huh? This should never compile; DIP1000 or not.

Comment (thewilsonator) [reply std/range/package.d]: Fixed.

**Comment by thewilsonator** [comment]
This is a master regression, so no stable.

**Comment by thewilsonator** [comment]
Stupid question: why does this fail, when the static assert in https://run.dlang.io/is/Vjjcjf doesn't?

**Comment by WalterBright** [comment]
Should remove `@trusted` from the declaration of `opAssign`

**Comment by thewilsonator** [comment]
Whoops! That'd certainly do it!

**Comment by thewilsonator** [comment]
Hmm, this seems to break https://github.com/dlang/phobos/blob/master/std/algorithm/iteration.d#L6181-L6201

on all except 
`"".substitute("","").equal("");` (line 6185)

`""w.substitute(""w,""w).equal(""w);` (line 6190), and 

`""d.substitute(""d,""d).equal(""d);` (line 6195)

**Comment by thewilsonator** [comment]
Seems to come from either https://github.com/dlang/phobos/blob/master/std/algorithm/iteration.d#L5791 or https://github.com/dlang/phobos/blob/master/std/algorithm/iteration.d#L5800

**Comment by berni44** [comment]
Is there any progress on this?

**Comment by thewilsonator** [comment]
Superseded by #7348

**Pull request closed without merging**